addons/token-exchange/secret_exchange_handler_register.go

```diff
@@ -0,0 +1,64 @@
+package addons
+
+import (
+	"fmt"
+
+	routev1 "github.com/openshift/api/route/v1"
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SecretExchangeHandler struct {
+	RegisteredHandlers map[string]SecretExchangeHandlerInerface
+}
+
+var secretExchangeHandler *SecretExchangeHandler
+
+// intialize secretExchangeHandler with handlers
+func registerHandler(spokeKubeConfig *rest.Config) error {
+	if secretExchangeHandler == nil {
+
+		// rook specific client
+		rookClient, err := rookclient.NewForConfig(spokeKubeConfig)
+		if err != nil {
+			return fmt.Errorf("failed to add rook client: %v", err)
+		}
```

Comment:
blue secret controller also creates a rook client. Let's reuse it.

Reply:
the client which i am creating from getClient is accessable by all handlers, no need to call getClient function everytime.

https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/60/commits/ac2c10b363be6f50164bf8615150394f035d9977#diff-f1703c8069bc1986c758619968f30e2a65647228557258ee7837bb41b6749a07R34

whenever new handler is created all they need to do is,  pass genericSpokeClient. so all struct function inside handlers will have this client. No need call getClient and creating new client everytime. Also, writting unit testing is easy. I can easily mock the client.

Reply:
i made blue and green secret controllers very simple. all logic are moved inside handlers. So register will takecare of intializing handlers with enough info.